### PR TITLE
Bug 1986253: ODC-5584 - Application Groupings

### PR DESCRIPTION
**Pull request by gajanan-more**
**Fix:** https://issues.redhat.com/browse/ODC-5584

**Problem:** Automation to perform actions on Application Groupings.

**Solution:** This script will automate the actions performing on Application Groupings.

**Test Setup:**
1. Cluster should be running on local
2. Check the TAGS in `frontend/packages/topology/integration-tests/cypress.json` file be: `"@Topology and (@smoke or @regression) and not (@Manual or @to-do or @un-verified)",`

**Commands to execute:**
1. `yarn run test-cypress-topology`
2. Execute the `topology-application-grouping.feature` file

**Checks for approving Automation PR:**
- [x]  Execute the @to-do tagged gherkin scripts manually
- [x]  Convert the @to-do gherkin scripts to cypress automation scripts
- [x]  Once scripts are automated, remove tag @to-do
- [x]  Execute the scripts in Remote cluster

**Browser**
Chrome 92

**Test Execution Screenshot:**
Screenshot 2021-07-23 at 2 02 54 PM

### CODE REVIEW

**Comment by makambalaji** [comment]
Please update the automation status report

**Comment by gajanan-more** [comment]
> Please update the automation status report

I have updated the Automation Status Report.

File: frontend/packages/topology/integration-tests/features/topology/topology-application-grouping.feature [hunk 1]
```diff
@@ -3,66 +3,41 @@ Feature: Application groupings in topology
...
-              And user clicks on Add to Project on Action menu and clicks on From Devfile
-              And user fills the form and clicks Create
-              And user clicks on Add to Project on Action menu and clicks on From Event Source
-              And user selects Api Server Source and clicks on Create Event Source
+            Given user is at Topology page
+             When user clicks on application groupings "nodejs-ex-git-app"
+              And user clicks on Action menu
+              And user clicks "Add to Application" from action menu
+              And user clicks on "From Git"
               And user fills the form and clicks Create
```
Comment (makambalaji) [inline frontend/packages/topology/integration-tests/features/topology/topology-application-grouping.feature:33]: Can you please specify the details including workload name

**Review by makambalaji — CHANGES_REQUESTED** [review]

File: frontend/packages/topology/integration-tests/support/pages/topology/topology-page.ts [hunk 2]
```diff
@@ -168,9 +173,32 @@ export const topologyPage = {
...
+      .click({ force: true });
+  },
+  verifyApplicationGroupingsDeleted: (appName: string) => {
+    const id = `[data-id="group:${appName}"]`;
+    cy.get(id, { timeout: 50000 }).should('not.exist');
+  },
+  verifyApplicationGroupings: (workloadName: string) => {
+    cy.get(topologyPO.sidePane.applicationGroupingsTitle).should('be.visible');
+    const id = `[data-test-id="${workloadName}"]`;
+    cy.get(id).should('be.visible');
```
Comment (sanketpathak) [inline frontend/packages/topology/integration-tests/support/pages/topology/topology-page.ts:200]: We can club this together as:
```suggestion
    cy.byTestID(workloadName)
```

File: frontend/packages/topology/integration-tests/support/pages/topology/topology-side-pane-page.ts [hunk 3]
```diff
@@ -122,6 +122,13 @@ export const topologySidePane = {
   verifyPipelineRuns: () => {
     cy.get(topologyPO.sidePane.resourcesTab.pipelineRuns).should('be.visible');
   },
+  verifyResourcesApplication: (deploymentName: string) => {
+    cy.get(`[data-test-id="${deploymentName}"]`).should('be.visible');
```
Comment (sanketpathak) [inline frontend/packages/topology/integration-tests/support/pages/topology/topology-side-pane-page.ts:126]:
```suggestion
    cy.byTestID(deploymentName).should('be.visible');
```

File: frontend/packages/dev-console/integration-tests/support/constants/topology.ts [hunk 4]
```diff
@@ -36,9 +36,21 @@ export enum nodeActions {
   DeleteInMemoryChannel = 'Delete InMemoryChannel',
   EditRevision = 'Edit Revision',
   DeleteRevision = 'Delete Revision',
+  DeleteApplication = 'Delete Application',
+  AddtoApplication = 'Add to Application',
```
[the remark this answers is not in the document]
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/support/constants/topology.ts:40]: Updated

File: frontend/packages/topology/integration-tests/features/topology/topology-application-grouping.feature (hunk 1 above)
Comment (gajanan-more) [reply frontend/packages/topology/integration-tests/features/topology/topology-application-grouping.feature:33]: Updated as `user fills the form with workload name "added-application-1" and clicks Create`

File: frontend/packages/topology/integration-tests/support/pages/topology/topology-page.ts (hunk 2 above)
Comment (gajanan-more) [reply frontend/packages/topology/integration-tests/support/pages/topology/topology-page.ts:200]: Updated

File: frontend/packages/topology/integration-tests/support/pages/topology/topology-side-pane-page.ts (hunk 3 above)
Comment (gajanan-more) [reply frontend/packages/topology/integration-tests/support/pages/topology/topology-side-pane-page.ts:126]: Updated

**Comment by sanketpathak** [comment]
Working as expected 
Screenshot from 2021-07-26 15-14-40
/lgtm

**Comment by makambalaji** [comment]
/lgtm
working fine on Chrome 92 browser
image

**Pull request merged**